### DISCUSSION TITLE
Do formulae with livecheck blocks automatically get updated from upstream?

### DISCUSSION THREAD

**Discussion by puredanger** [Homebrew/discussions#19] [Everyday usage] [answered]
Recently we released a new version of Clojure (1.10.1.697) in the upstream. 

* Upstream: https://github.com/clojure/homebrew-tools/blob/master/Formula/clojure.rb

The clojure formula in homebrew-core has a livecheck block looking at this:

* homebrew-core: https://github.com/homebrew/homebrew-core/blob/master/Formula/clojure.rb

I'm not sure what to expect with respect to livecheck - should this formula get automatically bumped? If so, at what frequency is it checked? Or is there something explicit to do? Just trying to figure out what my expectations should be.

Thanks...

**Comment by Rylan12** [comment, marked as answer]
To answer your question, `livecheck` _does not_ automatically update formulae. We rely on community contributions to keep packages up to date. You're strongly encouraged to open a PR to update the `clojure` formula. Take a look at [the docs](https://docs.brew.sh/Updating-Software-in-Homebrew) for a guide on how to do this ([`brew bump-formula-pr`](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request#submit-a-new-version-of-an-existing-formula) is probably what you're looking for)

`livecheck` is a tool that compares the version we have to the latest upstream version. For example, running `brew livecheck clojure` will indicate that there is a new version available:

```console
$ brew livecheck clojure
clojure : 1.10.1.561 ==> 1.10.1.697
```

There has been [some discussion](https://github.com/Homebrew/brew/issues/5725) about creating pull requests automatically based on `livecheck` but this hasn't been implemented yet.

**Reply by puredanger** [reply]
thanks!

**Comment by puredanger** [comment]
I did try to find any documentation that would answer this question but I did not find it (not sure if that was a problem with my finding or in the docs).

**Reply by Rylan12** [reply]
I don't think we have any docs about `livecheck` (aside from the [manpage entry](https://docs.brew.sh/Manpage#livecheck-formulae)). We do have extensive documentation about how to update our formulae, though.

Some key links:
- [Updating Software in Homebrew](https://docs.brew.sh/Updating-Software-in-Homebrew)
- [Adding Software to Homebrew](https://docs.brew.sh/Adding-Software-to-Homebrew)
- [How to open a pull request](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request)
- [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook)
- [homebrew/core contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
